router/client/client.go

```diff
@@ -50,13 +50,13 @@ func NewWithAddr(addr string) Client {
 type Client interface {
 	// CreateRoute creates a new route.
 	CreateRoute(*router.Route) error
-	// SetRoute updates an existing route. If the route does not exist, it
-	// creates a new one.
-	SetRoute(*router.Route) error
-	// DeleteRoute deletes the route with the specified id.
-	DeleteRoute(id string) error
-	// GetRoute returns a route with the specified id.
-	GetRoute(id string) (*router.Route, error)
+	// UpdateRoute updates an existing route by overwriting all fields on the route
+	// except ID and Domain.
+	UpdateRoute(*router.Route) error
+	// DeleteRoute deletes the route with the specified routeType and id.
+	DeleteRoute(routeType, id string) error
+	// GetRoute returns a route with the specified routeType and id.
+	GetRoute(routeType, id string) (*router.Route, error)
```

Comment:
What do you think about making `routeType` a custom type with constants for the valid options?